src/gui/menu_vars.h

```diff
@@ -50,17 +50,29 @@ struct MenuVars {
     constexpr static const size_t RANGE_SZ = 3;
     constexpr static const char *const zoffset_prt_format = "%.3f";
     constexpr static const char *const labels[] = { N_("Move X"), N_("Move Y"), N_("Move Z"), N_("Move E") };
-    static const std::array<std::array<int16_t, RANGE_SZ>, AXIS_CNT> axis_ranges;
+
+    static const std::array<int, AXIS_CNT> default_steps_per_unit;
+    static const std::array<int, AXIS_CNT> default_microsteps;
+    static const std::array<int, AXIS_CNT> default_currents;
+
+    static const std::array<int, MenuVars::RANGE_SZ> maximum_z_axis_range;
+    static const std::array<std::array<int, RANGE_SZ>, AXIS_CNT> axis_ranges;
+    static const std::array<std::array<int, RANGE_SZ>, AXIS_CNT> steps_per_unit_ranges;
+    static const std::array<std::array<int, RANGE_SZ>, AXIS_CNT> axis_rms_currents_ranges;
+
     static const int16_t manual_feedrate[AXIS_CNT];
     static const char axis_letters[AXIS_CNT];
     static const int16_t extrude_min_temp;
 
-    constexpr static std::array<uint8_t, RANGE_SZ> printfan_range = { 0, 255, 1 };
-    constexpr static std::array<uint16_t, RANGE_SZ> flowfact_range = { 50, 150, 1 };
-    constexpr static std::array<uint16_t, RANGE_SZ> feedrate_range = { 10, 255, 1 };
+    constexpr static std::array<int, RANGE_SZ> printfan_range = { 0, 255, 1 };
+    constexpr static std::array<int, RANGE_SZ> flowfact_range = { 50, 150, 1 };
+    constexpr static std::array<int, RANGE_SZ> feedrate_range = { 10, 255, 1 };
+    constexpr static std::array<int, MenuVars::RANGE_SZ> microstep_exponential_range = { 1, 256, 2 }; // 2^0 - 2^8 .. 1, 2, 4, .. , 128, 256
+    constexpr static std::array<int, MenuVars::RANGE_SZ> axis_rms_currents_range = { 0, 800, 1 };
+    constexpr static std::array<int, MenuVars::RANGE_SZ> steps_per_unit_range = { 1, 1000, 1 }; // small range, experimental feature
     //todo make constexpr
-    static const std::array<uint16_t, RANGE_SZ> nozzle_range;
-    static const std::array<uint8_t, RANGE_SZ> bed_range;
+    static const std::array<int, RANGE_SZ> nozzle_range;
```

Comment:
this can pose a potential problem in initialization order - nozzle_range is not constexpr, but it is used to initialize spinrange in menu_vars.cpp -> use some singleton to make sure the linker makes the initialization order correctly.